src/plus/integrations/providers/bitbucket/bitbucket-api.ts

```diff
@@ -0,0 +1,289 @@
+import { isWeb } from '@env/platform';
+import type { Container } from '../../../../container';
+import {
+	AuthenticationError,
+	AuthenticationErrorReason,
+	CancellationError,
+	ProviderFetchError,
+	RequestClientError,
+	RequestNotFoundError,
+} from '../../../../errors';
+import type { IssueOrPullRequest, IssueOrPullRequestType } from '../../../../git/models/issueOrPullRequest';
+import type { PullRequest } from '../../../../git/models/pullRequest';
+import type { Provider } from '../../../../git/models/remoteProvider';
+import { showIntegrationRequestFailed500WarningMessage } from '../../../../messages';
+import { configuration } from '../../../../system/-webview/configuration';
+import { debug } from '../../../../system/decorators/log';
+import { Logger } from '../../../../system/logger';
+import type { LogScope } from '../../../../system/logger.scope';
+import { getLogScope } from '../../../../system/logger.scope';
+import { maybeStopWatch } from '../../../../system/stopwatch';
+import type { BitbucketIssue, BitbucketPullRequest } from './models';
+import { bitbucketIssueStateToState, fromBitbucketPullRequest } from './models';
+
+export class BitbucketApi implements Disposable {
+	private readonly _disposable: Disposable;
+
+	constructor(_container: Container) {
+		this._disposable = configuration.onDidChangeAny(e => {
+			if (
+				configuration.changedCore(e, ['http.proxy', 'http.proxyStrictSSL']) ||
+				configuration.changed(e, ['outputLevel', 'proxy'])
+			) {
+				this.resetCaches();
+			}
+		});
+	}
+
+	dispose(): void {
+		this._disposable.dispose();
+	}
+
+	private _proxyAgent: HttpsProxyAgent | null | undefined = null;
+	private get proxyAgent(): HttpsProxyAgent | undefined {
+		if (isWeb) return undefined;
+
+		if (this._proxyAgent === null) {
+			this._proxyAgent = getProxyAgent();
+		}
+		return this._proxyAgent;
+	}
+
+	private resetCaches(): void {
+		this._proxyAgent = null;
+	}
+
+	@debug<BitbucketApi['getPullRequestForBranch']>({ args: { 0: p => p.name, 1: '<token>' } })
+	public async getPullRequestForBranch(
+		provider: Provider,
+		token: string,
+		owner: string,
+		repo: string,
+		branch: string,
+		options: {
+			baseUrl: string;
+		},
+	): Promise<PullRequest | undefined> {
+		const scope = getLogScope();
+
+		const response = await this.request<{
+			values: BitbucketPullRequest[];
+			pagelen: number;
+			size: number;
+			page: number;
+		}>(
+			provider,
+			token,
+			options.baseUrl,
+			`repositories/${owner}/${repo}/pullrequests?q=source.branch.name="${branch}"&fields=values.*`,
+			{
+				method: 'GET',
+			},
+			scope,
+		);
+
+		if (!response?.values?.length) {
+			return undefined;
+		}
+		return fromBitbucketPullRequest(response.values[0], provider);
+	}
+
+	@debug<BitbucketApi['getIssueOrPullRequest']>({ args: { 0: p => p.name, 1: '<token>' } })
+	public async getIssueOrPullRequest(
+		provider: Provider,
+		token: string,
+		owner: string,
+		repo: string,
+		id: string,
+		options: {
```

Comment:
An `options` param shouldn't be required. Any required bits should be outside of the options object.

```suggestion
		options?: {
```